tools/third_party.py

```diff
@@ -208,9 +209,47 @@ def download_from_google_storage(item, bucket):
         env=google_env())
 
 
+# Download the given item from Chrome Infrastructure Package Deployment.
+def download_from_cipd(item):
+    if sys.platform == 'win32':
+        root_dir = "v8/buildtools/win"
+        item += "windows-amd64"
+    elif sys.platform == 'darwin':
+        root_dir = "v8/buildtools/mac"
+        item += "mac-amd64"
+    elif sys.platform.startswith('linux'):
+        root_dir = "v8/buildtools/linux64"
+        item += "linux-amd64"
+
+    # init cipd if necessary
+    if not os.path.exists(path.join(tp(root_dir), ".cipd")):
+        run([
+            tp('depot_tools/cipd'),
+            'init',
+            tp(root_dir),
+            '-force',
+        ],
+            env=google_env())
+
+    # read gn CIPD version from the DEPS file
+    deps = open(tp('v8/DEPS'), 'rb').read()
+    version = re.search("'gn_version': '(git_revision:[0-9a-f]{40})'",
+                        deps).group(1)
```

Comment:
Although it's a bit less automated, it seems like it would be less brittle to have `version` simply be an argument to this function which is manual copied out of the `v8/DEPS` file. What do you think?

Reply:
But you would need to manually update it with every `v8` upgrade. And reading in the `DEPS` file to get the correct version isn't so much different from reading the `gn.sha1` file as before. Of course it's a bit more complicated to parse though I've to admit.

Reply:
I mean, I don't really intend to update clang-format or gn ever again if possible (even if V8 does)

Having to debug this regex at some point is my concern... What if the DEPS file format changes?

Reply:
Sure, I've pinned the version of `gn` to the version used in `v8` 7.9.8, if you don't intend to update it anyway (`v8` usually upgrades `gn` with every major version now).